applications/GeoMechanicsApplication/tests/test_submerged_construction_of_excavation.py

```diff
@@ -0,0 +1,205 @@
+        return self.model_width * (
+                self.unsaturated_clay_layer_thickness * self.unsaturated_unit_weight_of_clay
+                + self.saturated_clay_layer_thickness * self.saturated_unit_weight_of_clay
+                + self.saturated_sand_layer_thickness * self.saturated_unit_weight_of_sand
+        )
+
+    def calculate_weight_of_excavated_clay_upper_right(self):
+        return (
+                self.unsaturated_clay_layer_thickness
+                * self.excavation_width
+                * self.unsaturated_unit_weight_of_clay
+        )
+
+    def calculate_weight_of_excavated_clay_middle_right(self):
+        return (
+                self.excavated_middle_clay_layer_thickness
+                * self.excavation_width
+                * self.saturated_unit_weight_of_clay
+        )
+
+    def calculate_weight_of_excavated_clay_lower_right(self):
+        return (
+                self.excavated_lower_clay_layer_thickness
+                * self.excavation_width
+                * self.saturated_unit_weight_of_clay
+        )
+
+    def calculate_weight_of_water_after_second_excavation(self):
+        return (
+                self.excavated_middle_clay_layer_thickness
+                * self.excavation_width
+                * self.unit_weight_of_water
+        )
+
+    def calculate_weight_of_water_after_third_excavation(self):
+        return (
+                (
+                        self.excavated_middle_clay_layer_thickness
+                        + self.excavated_lower_clay_layer_thickness
+                )
+                * self.excavation_width
+                * self.unit_weight_of_water
+        )
+
+    def calculate_weight_of_diaphragm_wall(self):
+        return self.weight_of_diaphragm_wall * self.length_of_diaphragm_wall
+
+    def calculate_total_vertical_surface_load(self):
+        return self.distributed_surface_load * self.load_edge_length
+
+    def check_vertical_reaction(self, output_reader, project_path, stage, expected):
+        filename = f"{stage['name']}.post.res"
+        output_data = output_reader.read_output_from(
+            os.path.join(project_path, filename)
+        )
+        rel_tolerance = 0.005
+        self.assertAlmostEqual(
+            self.total_reaction_y_from_output_data(
+                output_data, stage["time"], self.BOTTOM_NODE_IDS
+            ),
+            expected,
+            places=None,
+            delta=rel_tolerance * expected,
+        )
+
+    def run_simulation_and_checks(self, sub_directory_name):
+        project_path = test_helper.get_file_path(
+            os.path.join("submerged_construction_of_excavation", sub_directory_name)
+        )
+
+        with context_managers.set_cwd_to(project_path):
+            model = Kratos.Model()
+            for stage in self.stage_info.values():
+                param_file = os.path.join("..", "common", f"{stage['name']}.json")
+                with open(param_file, "r") as f:
```

Comment:
If we'd use Python's `pathlib` (`from pathlib import Path`), we could rewrite this to:
```suggestion
                with open(Path("..") / "common" /  f"{stage['base_name']}.json", "r") as f:
```

Reply:
used Path